### PR TITLE
RFC: pushing down filters for joins

### PR DESCRIPTION
**Pull request by RaduBerinde**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/12430)

### CODE REVIEW

File: docs/RFCS/join_filters.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+SELECT * FROM
+  (SELECT * FROM l WHERE leftExpr)
+JOIN
+  (SELECT * FROM r WHERE rightExpr)
+ON combinedExpr
+```
+
+## LEFT OUTER JOIN
+
+In a left outer join, all rows from the left table ends up in the output in some
```
Comment (spencerkimball) [inline docs/RFCS/join_filters.md]: s/ends/end/

File: docs/RFCS/join_filters.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+SELECT * FROM
+  l LEFT OUTER JOIN (SELECT * from r WHERE onRight)
+ON (onLeft AND onCombined)
+WHERE (filterLeft AND filterRight AND filterCombined)
+```
+
+ - `filterLeft` can be pushed down to `l`:
+```sql
+SELECT * FROM
+  (SELECT * from l WHERE onLeft) JOIN r
```
Comment (spencerkimball) [inline docs/RFCS/join_filters.md]: s/onLeft/filterLeft/

**Comment by RaduBerinde** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/RFCS/join_filters.md, line 56 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12430#-KZ3eJfo7inm8HabeE67-r1-56:-KZ8DpkjuShxFGSH0zwa:b-dtkdrq) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c39cac75311d39e624de7fdd828ce7eb7f201f8/docs/RFCS/join_filters.md#L56)):*
<details><summary><i>Previously, spencerkimball (Spencer Kimball) wrote…</i></summary><blockquote>

s/ends/end/
</blockquote></details>

Done, thanks.

---

*[docs/RFCS/join_filters.md, line 78 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/12430#-KZ3eJfo7inm8HabeE67-r1-78:-KZ8DqAKSYl_BIrln94z:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5c39cac75311d39e624de7fdd828ce7eb7f201f8/docs/RFCS/join_filters.md#L78)):*
<details><summary><i>Previously, spencerkimball (Spencer Kimball) wrote…</i></summary><blockquote>

s/onLeft/filterLeft/
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/12430)*

**Comment by RaduBerinde** [comment]
@knz I think you looked over this, ok to merge?

**Comment by knz** [comment]
yes LGTM

**Pull request merged**